lib/compress.js

```diff
@@ -4556,6 +4556,7 @@ merge(Compressor.prototype, {
 
     function retain_top_func(fn, compressor) {
         return compressor.top_retain
+            && fn
             && fn._top
```

Comment:
nit: just doing the instanceof first would make the subsequent `fn.*` valid.

```suggestion
            && fn instanceof AST_Defun
            && fn._top
```

Reply:
done